lib/Widgets/Dialog.vala

```diff
@@ -0,0 +1,67 @@
+/*
+* Copyright 2021 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public class Granite.Dialog : Hdy.Window {
```

Comment:
This is probably controversial and adds a dependency that we probably only want until Gtk4. Since we control the contents at the bottom of the window we don't strictly need this for rounded corners. What this does give us is a window with no titlebar area messing with our layout and lets us drag from anywhere in the window.